File: bfs-web-app/types/jwks.ts
export interface JWK {
  kty: string // Key Type - "OKP" for Ed25519
  crv: string // Curve - "Ed25519"
  kid: string // Key ID
  alg: string // Algorithm - "EdDSA"
  use: string // Public Key Use - "sig"
  x: string // Base64url-encoded public key
}

export interface JWKS {
  keys: JWK[]
}

File: bfs-backend/internal/domain/jwks.go
package domain

// JWKS represents a JSON Web Key Set response
type JWKS struct {
	Keys []JWK `json:"keys"`
}

// JWK represents a JSON Web Key for Ed25519
type JWK struct {
	KeyType   string `json:"kty"` // Key Type - "OKP" for Ed25519
	Curve     string `json:"crv"` // Curve - "Ed25519"
	KeyID     string `json:"kid"` // Key ID
	Algorithm string `json:"alg"` // Algorithm - "EdDSA"
	Use       string `json:"use"` // Public Key Use - "sig"
	X         string `json:"x"`   // Base64url-encoded public key
}
